### ISSUE TITLE
Bug crossing lines in line/area chart

### ISSUE THREAD

**Issue by maksperskiy** [apache/superset#29854]
### Bug description

Line doesn't render between points if values crossed by X Axis.
![image](https://github.com/user-attachments/assets/0274ae01-8485-46f9-b975-fa6108fb04c1)
Data:
```sql
select * from (
values ('first', 2023, 172520663.69, 172520663.69),
('first', 2024, 96318876.44, 96318876.44),
('second', 2023, 27512555.11, 27512555.11),
('second', 2024, 32673437.12, 34673437.12),
('third', 2023, 12536226.56, 12536226.),
('third', 2024, 9236723.51, 9236723.51),
('fourth', 2023, 70255125.99, 15255125.99),
('fourth', 2024, 61255558.42, 32255558.42)
) s("type", year, score, amount)
```

Version: 4.0.0

### How to reproduce the bug

1. Create dataset (use SQL from description)
2. Create Line Chart with "total" as X-AXIS, "SUM(score)" as METRICS, "type" as DIMENSIONS

### Superset version

4.0.2

### Python version

I don't know

### Node version

I don't know

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [ ] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by rusackas** [comment]
I think dosu is on the right path here... in essence, the x axis is usually time, and we'd expect to see values for all series at each interval. In a case like that, we'd either assume the missing data point is either null and not draw it (like you see here) or you can use the Advanced Analytics to resample the data and enable 'zero imputation' to make the missing datapoints a zero. However, when using a numeric axis like this, I'm not sure we should make these same assumptions... it might make sense to just make all series contiguous lines like you're expecting. I'm sure there are use cases where people _won't_ want that behavior (i.e. if you're expecting a point for each series at each x axis tick) so maybe we'll need a "contiguous lines" checkbox or something.

**Comment by osullivryan** [comment]
Hello, I am also experiencing the same with my data set. While I have a date within my query to return data in a specific range my x-axis is not temporal. This can make line plots a bit hard to follow with the discontinuity. I think an option to “continue” or even some interpolations would be great options.
